src/components/organisms/EventModal/EventModal.tsx

```diff
@@ -37,7 +37,15 @@ export const EventModal: React.FC<EventModalProps> = ({
   });
 
   const eventRoom = useMemo<Room | undefined>(
-    () => eventVenue?.rooms?.find((room) => room.title === event.room),
+    () =>
+      eventVenue?.rooms?.find((room) => {
+        const noSlashUrl = room.url.endsWith("/")
+          ? room.url.slice(0, -1)
+          : room.url;
+        const [roomName] = noSlashUrl.split("/").slice(-1);
+
+        return event?.room?.endsWith(`${roomName}`);
+      }),
     [eventVenue, event]
   );
```

Comment:
The changes here probably should be wrapped up into a helper method somewhere in `utils/*`.

Also, are there more locations across the platform that have similar logic to this/would have the same sort of bug/issues? I believe there will be, so we should be ensuring we fix it correctly in all locations.

Reply:
Also, 'rooms'/'portals' can link to external URLs.. will that impact your logic here?

Reply:
We may also want to make use of `matchPath` from https://reactrouter.com/web/api/matchPath wrather than attempting to 'roll our own' with regards to extracting sub-sections of in-platform URLs, since this is the same logic/lib that our actual routing uses.

In doing so, we would want to extract the relevant paths into a `const` in `src/settings` or similar, and use that in both `AppRouter` and the `utils/*` function that ends up using `matchPath` for this logic.

Reply:
There are also a bunch of helpers/etc in `src/utils/url.ts` that may be relevant here.

Reply:
I've moved this logic into helpers in /src/utils/url.ts at 8b0a07c
I've been thinking about the matchPath and overall url approach and that got me to a conclusion that it might not work as we want here
since both room/event urls can be set without any url validation ( really any string that you can imagine ), the matchpath here won't be working properly when checking those urls, so that made me think that we should add some sort of validation to the room/event url field ( probably not in this task tho? ) 
also, I've added specific if clause for the external URL ( thank you ) , so now even if user adds an external URL to an event and eventRoom is not found, he will still be able to visit that url by clicking on the redirect btn

Reply:
> so that made me think that we should add some sort of validation to the room/event url field ( probably not in this task tho? )

I think we might already have a backlog issue for this, but if not, it would be good to create one.